### PR TITLE
Mark throwable methods as pure

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

**Comment by greg0ire** [comment]
Some of the failing tests look like they are cause by my changes, I have to look into that.

**Comment by muglug** [comment]
Thanks! Looks like the failures were just due to my earlier bad merge

File: src/Psalm/Internal/Stubs/CoreImmutableClasses.phpstub [hunk 1]
```diff
@@ -61,21 +91,33 @@ class Exception implements Throwable
 
     /**
      * @psalm-mutation-free
-     * @return int|string
      */
-    public final function getCode(){}
+    public final function getCode(): int {}
```
Comment (andrew-demb) [inline src/Psalm/Internal/Stubs/CoreImmutableClasses.phpstub:95]: `getCode()` can return string

Comment (greg0ire) [reply src/Psalm/Internal/Stubs/CoreImmutableClasses.phpstub:95]: Are you sure? https://3v4l.org/cJHtA

Comment (andrew-demb) [reply src/Psalm/Internal/Stubs/CoreImmutableClasses.phpstub:95]: Yes. PDO throw exceptions with string codes.

https://3v4l.org/lP7cb
https://www.php.net/manual/en/exception.getcode.php

Comment (greg0ire) [reply src/Psalm/Internal/Stubs/CoreImmutableClasses.phpstub:95]: Indeed. WTF…

Comment (greg0ire) [reply src/Psalm/Internal/Stubs/CoreImmutableClasses.phpstub:95]: See fix here: https://github.com/vimeo/psalm/pull/3175

**Pull request merged**